packages/core/contracts/financial-templates/common/FundingRatePayer.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "../funding-rate-store/interfaces/FundingRateStoreInterface.sol";
+import "./FeePayer.sol";
+
+
+/**
+ * @title FundingRatePayer contract.
+ * @notice Extends FeePayer by adding funding rate store payment functionality for any financial contract that needs
+ * to access a funding rate. Contract is abstract as each derived contract that inherits `FundingRatePayer` must
+ * implement `pfc()`.
+ */
+
+abstract contract FundingRatePayer is FeePayer {
+    /****************************************
+     *      FEE PAYER DATA STRUCTURES       *
+     ****************************************/
+
+    // Tracks the last block time when the fees were paid.
+    uint256 private lastPaymentTime;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event FundingRateFeesPaid(uint256 indexed fundingRateFee, uint256 indexed lateFee);
+
+    /****************************************
+     *              MODIFIERS               *
+     ****************************************/
+
+    // modifier that calls continuous fee paying methods.
+    // TODO: Prettier doesn't allow modifiers to be overridable?
+    // prettier-ignore
+    modifier fees override {
```

Comment:
cool, didn't realize you could override modifiers.

Reply:
Yeah neither does prettier apparently

Reply:
Lol

Reply:
Looks like they resolved this [issue](https://github.com/prettier-solidity/prettier-plugin-solidity/issues/279) and I upgraded `prettier-plugin-solidity` as a result.